python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -168,9 +170,12 @@ def _get_current_features(self, type):
         def point_indices(self):
             # Return a cupy.ndarray containing the index values that each
             # point belongs to.
+            """
             offsets = cp.arange(0, len(self.xy) + 1, 2)
             sizes = offsets[1:] - offsets[:-1]
             return cp.repeat(self._series.index, sizes)
+            """
+            return self._series.index
```

Comment:
The length of index can be different from the same as number of points right? Unless I misunderstood, this seems wrong to me.

Reply:
What this was doing was duplicating the index, so that for a buffer of points the `point_indices` would be `[0, 0, 1, 1, 2, 2, ..., n, n]`, then in the final logic I would `groupby` those indices, then do some comparison on that. It was very inefficient for points since they always have the same offset. For `pip` at least, only one result is returned for each point, each pair of xy coords stored in interleaved format.

Reply:
I don't think we'll ever have an operation that works on a level more granular than points, so keeping two indices, `[0, 0]` for example for a single point, is necessary. Two values in points = 1 index.

Reply:
Ok, thanks for explaining. I did not notice this is inside `GeoColumnAccessor`, and that `self._series` is the point series. But can you remind me where is this index computed?

Reply:
As the index for the `GeoSeries`, I think it will typically just be a `RangeIndex`. As we discuss this I realize that this approach may only work for homogeneous `Point` `GeoSeries`. If it is a heterogeneous `GeoSeries` then the index values won't all refer to `Point`s, and will produce undesirable results.

Reply:
I thought what you want here is not the RangeIndex, but the index of the point that can be used as group-by keys? For example:
```
Offset: [0, 2, 3, 4]
Points: [(0.0, 0.0), (1.0, 1.0), (2.0, 2.0), (3.0, 3.0)]
Index:  [0, 0, 1, 2]
```

Reply:
Yes, the RangeIndex becomes a regular series when it becomes a column of `result` before the groupby. Points don't need to be grouped, however, their indices are all singular. A pip result that uses a geometry other than `Point` needs to have the `n` points grouped into a single result, for point the result is always a single.

Reply:
Got it, thanks for explaining.